src/server.rs

```diff
@@ -304,11 +305,13 @@ async fn info(
 
 async fn classification_with_gen(
     State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
     WithRejection(Json(request), _): WithRejection<Json<models::GuardrailsHttpRequest>, Error>,
 ) -> Result<impl IntoResponse, Error> {
     let request_id = Uuid::new_v4();
     request.validate()?;
-    let task = ClassificationWithGenTask::new(request_id, request);
+    let filtered_headers = filter_header(&state.orchestrator.get_config(), headers).await;
+    let task = ClassificationWithGenTask::new(request_id, request, filtered_headers);
```

Comment:
Update per other changes.

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = ClassificationWithGenTask::new(request_id, request, headers);
```